internal/gatewayapi/validate.go

```diff
@@ -260,6 +268,80 @@ func (t *Translator) validateBackendServiceImport(backendRef *gwapiv1a2.BackendR
 	return true
 }
 
+func (t *Translator) validateBackendRefBackend(backendRef *gwapiv1a2.BackendRef, parentRef *RouteParentContext, resources *Resources,
+	backendNamespace string, route RouteContext, kind gwapiv1.Kind,
+) bool {
+	// TODO: support additional route kinds
+	routeStatus := GetRouteStatus(route)
+
+	if !t.BackendEnabled {
+		status.SetRouteStatusCondition(routeStatus,
+			parentRef.routeParentStatusIdx,
+			route.GetGeneration(),
+			gwapiv1.RouteConditionResolvedRefs,
+			metav1.ConditionFalse,
+			gwapiv1.RouteReasonUnsupportedValue,
+			fmt.Sprintf("Resource %s/%s of type Backend cannot be used since Backend is disabled in Envoy Gateway configuration",
+				NamespaceDerefOr(backendRef.Namespace, route.GetNamespace()), string(backendRef.Name)),
+		)
+		return false
+	}
+
+	if kind != KindHTTPRoute {
+		status.SetRouteStatusCondition(routeStatus,
+			parentRef.routeParentStatusIdx,
+			route.GetGeneration(),
+			gwapiv1.RouteConditionResolvedRefs,
+			metav1.ConditionFalse,
+			gwapiv1.RouteReasonUnsupportedValue,
+			fmt.Sprintf("Resource %s/%s of type Backend is not supported for %s routes", NamespaceDerefOr(backendRef.Namespace, route.GetNamespace()),
+				string(backendRef.Name), kind),
+		)
+		return false
+	}
+
+	backend := resources.GetBackend(backendNamespace, string(backendRef.Name))
+	if backend == nil {
+		status.SetRouteStatusCondition(routeStatus,
+			parentRef.routeParentStatusIdx,
+			route.GetGeneration(),
+			gwapiv1.RouteConditionResolvedRefs,
+			metav1.ConditionFalse,
+			gwapiv1.RouteReasonBackendNotFound,
+			fmt.Sprintf("Backend %s/%s not found", NamespaceDerefOr(backendRef.Namespace, route.GetNamespace()),
+				string(backendRef.Name)),
+		)
+		return false
+	}
+
+	var unsupportedRef bool
+
+	for _, bep := range backend.Spec.Endpoints {
+		switch {
+		case bep.Unix != nil:
+			unsupportedRef = true
+		case bep.FQDN != nil && bep.FQDN.Hostname == "localhost":
+			unsupportedRef = true
+		case bep.IPv4 != nil && bep.IPv4.Address == "127.0.0.1":
```

Comment:
Even though the field is called `IPv4`, there's nothing that really prevents IPv6 addresses from being specified here as far as I can tell.

Consider also rejecting the IPv6 localhost address ? `::1` ?

Reply:
Also, localhost on Linux systems is not just `127.0.0.1`. It's `127.0.0.1/8`, which means that this check is wrong.

The correct way to check this would be to check if the provided address is contained in `127.0.0.0/8`, not to string-compare to `127.0.0.1`.

Reply:
yea, good point, we just discussed this in slack. I'll update the validation to comply with the K8s one for EP Slices that checks for the entire range.